tests/bud.bats

```diff
@@ -534,6 +534,114 @@ _EOF
+RUN --mount=type=cache,target=/mycache,uid=100,gid=101,z mkdir -p /mycache/foo/bar
+RUN --mount=type=cache,target=/mycache,uid=100,gid=101,z cat /mycache/somefile
+_EOF
+
+  # build should be successful since right `uid` and `gid` is used.
+  run_buildah build $WITH_POLICY_JSON -t source -f ${TEST_SCRATCH_DIR}/bud/platform/Dockerfile ${TEST_SCRATCH_DIR}/bud/platform
+  expect_output --substring "something"
+
+  # verify the default use-case
+  cat > ${TEST_SCRATCH_DIR}/bud/platform/Dockerfile << _EOF
+FROM alpine
+COPY somefile .
+RUN --mount=type=cache,target=/mycache,z cp somefile /mycache/somefile
+RUN --mount=type=cache,target=/mycache,z mkdir -p /mycache/foo/bar
+RUN --mount=type=cache,target=/mycache,z cat /mycache/somefile
+_EOF
+
+  # build should be successful
+  run_buildah build $WITH_POLICY_JSON -t source -f ${TEST_SCRATCH_DIR}/bud/platform/Dockerfile ${TEST_SCRATCH_DIR}/bud/platform
+  expect_output --substring "something"
+
+  # ----- Verify failure for bad use-cases --- #
+
+  cat > ${TEST_SCRATCH_DIR}/bud/platform/Dockerfile << _EOF
+FROM alpine
+RUN addgroup -S appgroup && adduser -S appuser -G appgroup
+USER 100:101
+RUN --mount=type=cache,target=/mycache,z touch /mycache/somefile
+RUN --mount=type=cache,target=/mycache,z mkdir -p /mycache/foo/bar
+RUN --mount=type=cache,target=/mycache,z cat /mycache/somefile
+_EOF
+
+  # build should fail with permission denied since container's user is not root
+  # and cache is not configured with `uid`,`gid` or `U`
+  run_buildah 1 build $WITH_POLICY_JSON -t source -f ${TEST_SCRATCH_DIR}/bud/platform/Dockerfile ${TEST_SCRATCH_DIR}/bud/platform
+  expect_output --substring "Permission denied"
+
+  cat > ${TEST_SCRATCH_DIR}/bud/platform/Dockerfile << _EOF
+FROM alpine
+RUN addgroup -S appgroup && adduser -S appuser -G appgroup
+USER 100:101
+RUN --mount=type=cache,target=/mycache,uid=200,gid=201,z touch /mycache/somefile
+RUN --mount=type=cache,target=/mycache,uid=200,gid=201,z mkdir -p /mycache/foo/bar
+RUN --mount=type=cache,target=/mycache,uid=200,gid=201,z cat /mycache/somefile
+_EOF
+
+  # build should fail with permission denied since container's user is not root
+  # and cache is not configured with `uid`,`gid` or `U`
+  run_buildah 1 build $WITH_POLICY_JSON -t source -f ${TEST_SCRATCH_DIR}/bud/platform/Dockerfile ${TEST_SCRATCH_DIR}/bud/platform
+  expect_output --substring "Permission denied"
+
+  # at last run a valid case again to verify if we not messed up central cache
+  cat > ${TEST_SCRATCH_DIR}/bud/platform/Dockerfile << _EOF
+FROM alpine
+COPY somefile .
+RUN --mount=type=cache,target=/mycache,z cp somefile /mycache/somefile
+RUN --mount=type=cache,target=/mycache,z mkdir -p /mycache/foo/bar
+RUN --mount=type=cache,target=/mycache,z cat /mycache/somefile
+_EOF
+
+  # build should be successful
+  run_buildah build $WITH_POLICY_JSON -t source -f ${TEST_SCRATCH_DIR}/bud/platform/Dockerfile ${TEST_SCRATCH_DIR}/bud/platform
+  expect_output --substring "something"
```

Comment:
Can you add a test to make sure the conflicts are in place.

uid=123,U

Reply:
Added a new test `build with --mount=type=cache: while parsing uid or gid must conflict with U"` which checks parsing use-cases.